src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/VsPowerShellHostTelemetryEmit.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.VisualStudio.Telemetry
+{
+    [Export(typeof(VsPowerShellHostTelemetryEmit))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class VsPowerShellHostTelemetryEmit : VsInstanceTelemetryConsts
```

Comment:
VsPowerShellHostTelemetryEmit seems very, very specific, feels like it could be designed to be a bit more generic.

Reply:
It reallly feels like there are too many classes. I'm not sure how to follow them.

Reply:
Powershell usage is unlike any other component in VS, it has many edge cases I believe it deserves own dedicated emitter. 
Not only it can be used between different solution loads, it can be used before load any solution and even after all solutions are closed. I think we can improve later if our requirement and usage changes in the future.